### DISCUSSION TITLE
Wallet connect doesnt sign message with signTypedData

### DISCUSSION THREAD

**Discussion by FurkanSezal** [WalletConnect/walletconnect-monorepo#3274] [Developer Support] [answered]
Hello i have a simple next js project. When i click the button i want to sign the message with my wallet. The code works perfectly with injected providers or metamask. But i couldnt work with walletconnect. 
I also tried with rainbowkit as wallet provider, i got same error with that. 
Here is error: 

    Unhandled Runtime Error
    MethodNotFoundRpcError: The method does not exist / is not available.
    
    Details: the method eth_signTypedData_v4 does not exist/is not available
    Version: viem@1.4.1

My app.js : 

    import {
      EthereumClient,
      w3mConnectors,
      w3mProvider,
    } from "@web3modal/ethereum";
    import { Web3Modal } from "@web3modal/react";
    import { configureChains, createConfig, WagmiConfig } from "wagmi";
    import { bscTestnet, mainnet } from "wagmi/chains";
    
    const chains = [bscTestnet, mainnet];
    const projectId = "PROJECT_ID";
    
    const { publicClient } = configureChains(chains, [w3mProvider({ projectId })]);
    const wagmiConfig = createConfig({
      autoConnect: true,
      connectors: w3mConnectors({ projectId, chains }),
      publicClient,
    });
    const ethereumClient = new EthereumClient(wagmiConfig, chains);
    
    export default function App({ Component, pageProps }) {
      return (
        <>
          <WagmiConfig config={wagmiConfig}>
            <Component {...pageProps} />;
          </WagmiConfig>
          <Web3Modal projectId={projectId} ethereumClient={ethereumClient} />
        </>
      );
    }

Here is my index.js: 

    import Head from "next/head";
    import Image from "next/image";
    import { Inter } from "next/font/google";
    import { getContract, parseEther } from "viem";
    import { abi } from "../comp/abi";
    import { Web3Button } from "@web3modal/react";
    import { useWeb3Modal } from "@web3modal/react";
    
    import {
      usePublicClient,
      useWalletClient,
      useAccount,
      useNetwork,
    } from "wagmi";
    
    export default function Home() {
      const walletClient = useWalletClient();
      const { address, isConnecting, isDisconnected } = useAccount();
    
      const { open, close } = useWeb3Modal();
    
      async function handleClick() {
        const signature = await walletClient.data.signTypedData({
          account: address,
          domain: {
            name: "Ether Mail",
            version: "1",
            chainId: 97,
            verifyingContract: "0x0000000000000000000000000000000000000000",
          },
          types: {
            EIP712Domain: [
              {
                name: "name",
                type: "string",
              },
              {
                name: "version",
                type: "string",
              },
              {
                name: "chainId",
                type: "uint256",
              },
              {
                name: "verifyingContract",
                type: "address",
              },
            ],
            Person: [
              { name: "name", type: "string" },
              { name: "wallet", type: "address" },
            ],
            Mail: [
              { name: "from", type: "Person" },
              { name: "to", type: "Person" },
              { name: "contents", type: "string" },
            ],
          },
          primaryType: "Mail",
          message: {
            from: {
              name: "Cow",
              wallet: "0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826",
            },
            to: {
              name: "Bob",
              wallet: "0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB",
            },
            contents: "Hello, Bob!",
          },
        });
    
        console.log(signature);
      }
    
      return (
        <>
          <div>
            <Web3Button></Web3Button>
          </div>
          <button onClick={() => open()}>Connect</button>
          <Head>
            <title>Create Next App</title>
            <meta name="description" content="Generated by create next app" />
            <meta name="viewport" content="width=device-width, initial-scale=1" />
            <link rel="icon" href="/favicon.ico" />
          </Head>
          <button onClick={handleClick}>Hello</button>
        </>
      );
    }

**Comment by glitch-txs** [comment]
Hi, which wallet are you testing with?

**Reply by FurkanSezal** [reply]
safePal

**Reply by FurkanSezal** [reply]
Matemask and rainbow wallet works with connect wallet but trust wallet and safepal wallet doesnt work with connect wallet. I got this error with trust wallet as well 
```
buildRequest.js:88 Uncaught (in promise) UnknownRpcError: An unknown RPC error occurred.

Details: Unknown method(s) requested
Version: viem@1.4.1
```

**Reply by glitch-txs** [reply, marked as answer]
For safePal is a known issue, we'll communicate with them to sort it out! For Trust if you're on iOS this should be resolved in the next update of the app.
